pkg/apis/v1alpha1/subnet_types.go

```diff
@@ -0,0 +1,78 @@
+/* Copyright © 2022 VMware, Inc. All Rights Reserved.
+   SPDX-License-Identifier: Apache-2.0 */
+
+package v1alpha1
+
+import (
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+type SubnetStatusCondition string
+type AccessMode string
+
+const (
+	SubnetReady    SubnetStatusCondition = "Ready"
+	SubnetNotReady SubnetStatusCondition = "NotReady"
+)
+
+// SubnetSpec defines the desired state of Subnet.
+type SubnetSpec struct {
+	// Size of Subnet based upon estimated workload count.
+	// Defaults to 64.
+	// +kubebuilder:default:=64
+	IPv4SubnetSize int `json:"ipv4SubnetSize,omitempty"`
+	// Access mode of Subnet, accessible only from within VPC or from outside VPC.
+	// Defaults to private.
+	// +kubebuilder:default:=private
+	// +kubebuilder:validation:Enum=private;public;isolated
+	AccessMode AccessMode `json:"accessMode,omitempty"`
+	// Subnet CIDRS.
+	// +kubebuilder:validation:MinItems=0
+	// +kubebuilder:validation:MaxItems=2
+	IPAddresses []string `json:"ipAddresses,omitempty"`
+}
+
+// SubnetCondition defines condition of Subnet.
+type SubnetCondition struct {
+	// Type defines condition type.
+	Type SubnetStatusCondition `json:"type"`
+	// Status defines status of condition type, True or False.
+	Status corev1.ConditionStatus `json:"status"`
+	// Reason shows a brief reason of condition.
+	Reason string `json:"reason,omitempty"`
+	// Message shows a human readable message about condition.
+	Message string `json:"message,omitempty"`
+}
+
+// SubnetStatus defines the observed state of Subnet.
+type SubnetStatus struct {
+	NSXTSubnetPath string            `json:"nsxtSubnetPath"`
```

Comment:
I discussed this field with Jianjun and Xiaopei, we prefer to change it to `NSXPath`, then all CRDs can use the same field name, which looks more common. What do you think?

Reply:
Sure, I'm ok with this as there is no other nsx resource path in this crd. Change it to `NSXPath` in `SubnetStatus` and `SubnetsetStatus`. Thanks.

Reply:
I think NSXAPIPath or NSXPolicyAPIPath is clearer. But let us choose one and use that for all CRDs.

Reply:
Ok, I'll align it with the name in struct VPCStatus(guess that it'd be `NSXPolicyPath`).

https://github.com/vmware-tanzu/nsx-operator/pull/106/files#r971438788